presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -283,11 +285,18 @@ protected Type visitQualifiedNameReference(QualifiedNameReference node, Analysis
                 // TODO This is kind of hacky, instead we should change the way QualifiedNameReferences are parsed
                 return tryVisitRowFieldAccessor(node);
             }
-            if (matches.size() > 1) {
+            if (matches.size() > 2 ||
+                    (matches.size() == 2
+                    && equivalentJoinFields.get(matches.get(0)) == null
+                    && equivalentJoinFields.get(matches.get(1)) == null)) {
                 throw new SemanticException(AMBIGUOUS_ATTRIBUTE, node, "Column '%s' is ambiguous", node.getName());
             }
 
-            Field field = Iterables.getOnlyElement(matches);
+            // if the field appears in join using, then find the corresponding base table's output field
+            Field field = matches.get(0);
+            if (matches.size() == 2 && equivalentJoinFields.get(matches.get(1)) == matches.get(0)) {
+                field = matches.get(1);
+            }
```

Comment:
The logic here is not quite straighforward. The first `if` does not check that one of `equivalentJoinFields.get(matches.get(1)) == matches.get(0)`, `(equivalentJoinFields.get(matches.get(0)) == matches.get(1))` is true. But the second `if` assumes so.

I would recommend:
```
Field field = null;
if (matches.size() == 1) {
    field = Iterables.getOnlyElement(matches);
}
else if (matches.size() == 2) {
    if (equivalentJoinFields.get(matches.get(1)) == matches.get(0)) {
        field = matches.get(1);
    }
    else if (equivalentJoinFields.get(matches.get(0)) == matches.get(1)) {
        field = matches.get(0);
    }
}
if (field == null) {
    throw new SemanticException(AMBIGUOUS_ATTRIBUTE, node, "Column '%s' is ambiguous", node.getName());
}
```